hack/make/test-docker-py

```diff
@@ -3,18 +3,47 @@ set -e
 
 source hack/make/.integration-test-helpers
 
-# subshell so that we can export PATH without breaking other things
+# The commit or tag to use for testing
+# TODO docker 17.06 cli client used in CI fails to build using a sha;
+# unable to prepare context: unable to 'git clone' to temporary context directory: error fetching: error: no such remote ref ac922192959870774ad8428344d9faa0555f7ba6
+#: exit status 128
+: ${DOCKER_PY_COMMIT:=3.7.0}
 (
 	bundle .integration-daemon-start
 
-	dockerPy='/docker-py'
-	[ -d "$dockerPy" ] || {
-		dockerPy="$DEST/docker-py"
-		git clone https://github.com/docker/docker-py.git "$dockerPy"
-	}
+	docker_host_scheme=`echo "${DOCKER_HOST}" | cut -d: -f1 -`
 
-	# exporting PYTHONPATH to import "docker" from our local docker-py
-	test_env PYTHONPATH="$dockerPy" py.test --junitxml="$DEST/results.xml" "$dockerPy/tests/integration"
+	case "${docker_host_scheme}" in
+		unix)
+			# trim the tcp:// scheme, and bind-mount the docker socket into the container
+			run_opts="-v ${DOCKER_HOST#unix://}:/var/run/docker.sock"
+			;;
 
+		tcp)
+			# run container in host-mode networking so that it can connect to the
+			# daemon from the current networking namespace (e.g., to connect to localhost)
+			run_opts=--network=host -e DOCKER_HOST=${DOCKER_HOST}
```

Comment:
Doesn't this need quotes to prevent Bash from trying to run `-e` as a command?

Reply:
Good catch; I think I moved the `-e` here after initial testing with a remote daemon 😊